src/org/ods/usecase/LeVADocumentUseCase.groovy

```diff
@@ -151,32 +151,32 @@ class LeVADocumentUseCase extends DocGenUseCase {
     }
 
     String createCSD(Map repo = null, Map data = null) {
-        // TODO Crete full reimplementation for the union of CS, FS and URS.
-        // See old commits in order to obtain the information on how those documents were implemented
         def documentType = DocumentType.CSD as String
 
         def sections = this.jiraUseCase.getDocumentChapterData(documentType)
         if (!sections) {
             throw new RuntimeException("Error: unable to create ${documentType}. Could not obtain document chapter data from Jira.")
         }
 
-        def interfaces = this.project.getSystemRequirementsTypeInterfaces().collect { req ->
-            [
-                    key        : req.configSpec.key,
-                    // TODO: change ur_key to req_key in template
-                    req_key    : req.key,
-                    description: req.description
-            ]
-        }
-
-        if (!sections."sec4") sections."sec4" = [:]
-        sections."sec4".items = SortUtil.sortIssuesByProperties(interfaces, ["req_key", "key"])
+        def requirements = this.project.getRequirementsGroupByGAMPTopic().collectEntries{ k, v -> 
```

Comment:
@martsec in class Project, we have a method `getSystemRequirements` that accepts an optional `componentName` argument. If you want to extend Project, then please continue to support the existing pattern (both naming and filtering of component name). Otherwise, you could also inline your implementation of `getRequirementsGroupByGAMPTopic` here and refactor only if there's a wider uses for it to bring it into Project. I leave this up to you.

Reply:
Code adapted to use the getSystemRequirements method